src/internal/functions/Get-TypeParameterContent.ps1

```diff
@@ -0,0 +1,54 @@
+Function Get-TypeParameterContent {
+  <#
+  .SYNOPSIS
+    Return the text for a Puppet Resource API Type given a DSC Resouce.
+  .DESCRIPTION
+    Return the text for a Puppet Resource API Type given a DSC Resouce.
+    It will return the text but _not_ directly write out the file.
+  .PARAMETER ParameterInfo
+    A DSCResourceInfo object with the required parameter information
+    retrieved - if the object does _not_ already have the updated info
+    for the parameters, this function will attempt to retrieve it.
+  .EXAMPLE
+    Get-DscResourceTypeInformation -Name PSRepository | Get-TypeParameterContent
+
+    This will retrieve a DSC resource from the PSModulePath, retrieve the information
+    needed to represent the DSC resource's properties as Puppet Resource API type
+    attributes and then return the representation of that DSC resource appropriate
+    for the Puppet Resource API.
+  #>
+  [cmdletbinding()]
+  Param (
+    [OutputType([String], [String[]])]
+    [pscustomobject]$ParameterInfo
```

Comment:
This function says it's input comes from Get-DscResourceInformation as does Get-TypeContent, but the parameter type specified here is different from that function. Is that necessary? Just not sure what type of object that cmdlet is actually putting out at this point.

Reply:
The `ParameterInfo` property is a NoteProperty added by `Get-DscResourceTypeInformation` whose value is an array of PSCustomObjects, but the only way to get that info is via the function or to manually construct an example object (as we do in the tests).

Reply:
Ahhh, I forgot about that.